File: GMS-backend/GMS/Entities/Endpoint/Teretana/Edit/TeretaneEditEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using Microsoft.AspNetCore.Mvc;

namespace GMS.Entities.Endpoint.Teretana.Edit
{
    [Route("Teretana-Edit")]

    public class TeretaneEditEndpoint : MyBaseEndpoint<TeretaneEditRequest, int>
    {
        private readonly ApplicationDbContext db;

        public TeretaneEditEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpPost]
        public override async Task<int> Handle([FromBody]TeretaneEditRequest request, CancellationToken cancellationToken)
        {
            Models.Teretana? teretana;
            if (request.ID == 0)
            {
                teretana = new Models.Teretana();
                db.Add(teretana);


            }
            else
            {
                teretana = db.Teretana.FirstOrDefault(s => s.ID == request.ID);
                if (teretana == null)
                    throw new Exception("pogresan ID");
            }

            teretana.Naziv = request.Naziv.RemoveTags();
            teretana.Adresa = request.Adresa.RemoveTags();
            teretana.GradID = request.GradID;


            await db.SaveChangesAsync(cancellationToken);

            return teretana.ID;
        }
    }
}

File: GMS-frontend/src/app/endpoints/teretane-endpoints/teretane-edit-endpoint.ts

import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {Config} from "../../config";
import {Observable} from "rxjs";
@Injectable({providedIn: 'root'})
export class TeretaneEditEndpoint implements  MyBaseEndpoint<TeretaneEditRequest, number>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: TeretaneEditRequest): Observable<number> {
    let url=Config.adresa + `Teretana-Edit`;

    return this.httpClient.post<number>(url, request);
  }

}
export interface TeretaneEditRequest {
  id: number
  naziv: string
  gradID: number
  adresa: string
}

